packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -228,3 +229,16 @@ export const updateEntity: ResolverFn<
 
   return mapEntityModelToGQL(updatedEntityModel);
 };
+
+export const archiveEntity: ResolverFn<
+  Promise<boolean>,
+  {},
+  LoggedInGraphQLContext,
+  MutationArchiveEntityArgs
+> = async (_, { entityId }, { dataSources: { graphApi }, userModel }) => {
+  const entityModel = await EntityModel.getLatest(graphApi, { entityId });
+
+  await entityModel.archive(graphApi, { actorId: userModel.getEntityUuid() });
+
+  return true;
```

Comment:
What does it mean to return a boolean here? I assume we're throwing an error on the failure case (as I don't see a `return false`) so would we perhaps want to return `null` or something if that's even possible?

Reply:
Yes we are throwing an error. GraphQL does not have a `void` type so this seemed like the easiest thing to do. GraphQL also [doesn't seem to support returning `null` as you suggest](https://github.com/ardatan/graphql-tools/issues/277#issuecomment-279144247). It would be possible to introduce a scalar called `Void` if you feel strongly about not using `Boolean`.

Reply:
Oh right it doesn't support null, remember talking about this a long time ago. Boolean seems fine for now